zio-schema/shared/src/main/scala/zio/schema/StandardType.scala

```diff
@@ -99,6 +105,135 @@ object StandardType {
     override def tag                                  = Tags.STRING
     override def compare(x: String, y: String): Int   = x.compareTo(y)
     override def defaultValue: Either[String, String] = Right("")
+
+    override def coerce[B](that: StandardType[B]): Option[String => Either[String, B]] =
+      that match {
+        case UnitType   => None
+        case StringType => Some(a => Right(a))
+        case BoolType   => Some(a => a.toBooleanOption.toRight(s"String $a could not be coerced into a Boolean"))
+        case ShortType  => Some(a => a.toShortOption.toRight(s"String $a could not be coerced into a Short"))
+        case IntType    => Some(a => a.toIntOption.toRight(s"String $a could not be coerced into an Int"))
+        case LongType   => Some(a => a.toLongOption.toRight(s"String $a could not be coerced into a Long"))
+        case FloatType  => Some(a => a.toFloatOption.toRight(s"String $a could not be coerced into a Float"))
+        case DoubleType => Some(a => a.toDoubleOption.toRight(s"String $a could not be coerced into a Double"))
+        case BinaryType => Some(a => Right(Chunk.fromArray(a.getBytes)))
+        case CharType   => None
+        case UUIDType =>
+          Some(
+            a =>
+              scala.util
+                .Try(java.util.UUID.fromString(a))
+                .toOption
+                .toRight(s"String $a could not be coerced into a UUID")
+          )
+        case BigDecimalType =>
+          Some(
+            a =>
+              scala.util
+                .Try(BigDecimal(a).bigDecimal)
+                .toOption
+                .toRight(s"String $a could not be coerced into a BigDecimal")
+          )
+        case BigIntegerType =>
+          Some(
+            a =>
+              scala.util
+                .Try(BigInt(a).bigInteger)
+                .toOption
+                .toRight(s"String $a could not be coerced into a BigInteger")
+          )
+        case DayOfWeekType =>
+          Some(
+            a =>
+              scala.util.Try(DayOfWeek.valueOf(a)).toOption.toRight(s"String $a could not be coerced into a DayOfWeek")
+          )
+        case MonthType =>
+          Some(
+            a => scala.util.Try(Month.valueOf(a)).toOption.toRight(s"String $a could not be coerced into a Month")
```

Comment:
For performance reasons, maybe good to have an `EitherUtil.attemptOrElse[A](a: => A, msg: E): Either[E, A]`.